xGame/Java/xgame/src/main/java/com/xgame/common/viewmodels/MatchHistoryViewModel.java

```diff
@@ -0,0 +1,86 @@
+package com.xgame.common.viewmodels;
+
+import java.sql.Timestamp;
+
+import com.xgame.common.enums.MatchOutcome;
+import com.xgame.data.entities.ChessMatch;
+import com.xgame.data.entities.User;
+import com.xgame.service.engine.ChessPiece.Color;
+
+public class MatchHistoryViewModel {
+	
+	private Timestamp startTime;
+	private Timestamp endTime;
+	private String opponentNickname;
+	private Color color;
+	private MatchOutcome outcome;
+	private Integer moveCount;
+	private String winningPlayer = null;
+	
+	public MatchHistoryViewModel(ChessMatch match, User user) {
+		MatchViewModel matchView = new MatchViewModel(match);
+		this.startTime = match.getStartTimestamp();
+		this.endTime = match.getLastTurnTimestamp();
+		this.color = matchView.getBlackPlayerId() == user.getId() ? Color.BLACK : Color.WHITE;
+		this.opponentNickname = this.color == Color.BLACK ? matchView.getWhitePlayerNickname() : matchView.getBlackPlayerNickname();
+		this.outcome = match.getMatchOutcome();
+		if(this.outcome != MatchOutcome.DRAW) this.setWinningPlayer(match.getWinningPlayer().getNickname());
```

Comment:
Instead, get the winningPlayer and null check it. WinningPlayer will only ever have a value if the match is over and there was a victory by either player. Also, don't call the setter, just set the value directly.

Reply:
Not sure what you mean by the first part, getting and null checking. Could you clarify?

Reply:
The ChessMatch entity has a reference to the winning player. So it makes more sense to check if there is a winning player, instead of checking the MatchOutcome. So

`If(match.getWinningPlayer() != null)`

instead of 

`if(this.outcome != MatchOutcome.DRAW)`

Reply:
Never mind. I misunderstood the intent of this check.